server/app/controllers/JSONFormats.scala

```diff
@@ -7,6 +7,7 @@ trait JSONFormats {
   private val lineFormat = Json.format[Line]
   private val testResultFormat = Json.format[TestResult]
   private val compilingFormat = Json.format[Compiling]
+  private val solutionTemplateFormat = Json.format[SolutionTemplate]
```

Comment:
Presence of this class is a bit weird. I think Play should able to use Typeclasses to write and read JSON.